spring-5-webflux/src/test/java/com/baeldung/spring/serverconfig/TimeoutLiveTest.java

```diff
@@ -0,0 +1,88 @@
+package com.baeldung.spring.serverconfig;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import io.netty.channel.ChannelOption;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.netty.handler.timeout.ReadTimeoutException;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import io.netty.handler.timeout.WriteTimeoutHandler;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.TcpClient;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@DirtiesContext
+public class TimeoutLiveTest {
+
+    private static final String BASE_URL = "https://localhost:8443";
+    private static final int TIMEOUT_MILLIS = 2000;
+
+    private WebTestClient webTestClient;
+
+    @Before
+    public void setup() throws SSLException {
+        webTestClient = WebTestClient.bindToServer(getConnector())
+                                     .baseUrl(BASE_URL)
+                                     .build();
+    }
+
+    @Test
+    public void shouldTimeout() {
+        try {
+            webTestClient.get()
+              .uri("/timeout/{timeout}", 3)
+              .exchange();
+            Assert.fail();
+        } catch (ReadTimeoutException ignore) {
+        }
+    }
+
+    @Test
+    public void shouldNotTimeout() {
+        try {
+            WebTestClient.ResponseSpec response = webTestClient.get()
+              .uri("/timeout/{timeout}", 1)
+              .exchange();
+            response.expectStatus()
+              .isOk()
+              .expectBody(String.class)
+              .isEqualTo("OK");
+        } catch (ReadTimeoutException e) {
```

Comment:
I would remove this since the exception fails the test, and it's better to have that reported then to hide it

Reply:
Changes have been made.